libevm/JitVM.cpp

```diff
@@ -248,47 +242,112 @@ int64_t evm_call(
 	return cost;
 }
 
+
+/// RAII wrapper for an evm instance.
+class EVM
+{
+	evm_instance* m_instance = nullptr;
+
+public:
+	EVM(evm_query_fn _queryFn, evm_update_fn _updateFn, evm_call_fn _callFn)
+	{
+		m_instance = evm_create(_queryFn, _updateFn, _callFn);
+	}
+
+	~EVM()
+	{
+		evm_destroy(m_instance);
+	}
+
+	EVM(EVM const&) = delete;
+	EVM& operator=(EVM) = delete;
```

Comment:
Here it does not matter. But I like implementing `operator=` with the copy&swap idiom. The implementation exploits that the param is passed by value (copied with the copy constructor) and that copy can be modified freely inside `operator=`, i.e. you can swap it with `this`.